### PR TITLE
Add Mac OS build setup steps

### PR DESCRIPTION
**Pull request by kyle-rader**
These steps are the outcome of getting a working build on a Mac pro (Late 2013) with a freshly installed 

```
macOS High Sierra 10.13.6
```

Updates:
* Add a Build setup step to the main README.md
* Remove the macOS sdk version from the ProjFS build scripts. (Because XCode merrrrrrr removes SDKs at will...)

Questions:
* What to say about the Apple Developer/Cert Signing step?

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -46,15 +46,95 @@ If you'd like to build your own VFS for Git Windows installer:
...
 The installer can now be found at `C:\Repos\VFSForGit\BuildOutput\GVFS.Installer\bin\x64\[Debug|Release]\SetupGVFS.<version>.exe`
 
+## Building VFS for Git on Mac
+
+### First setup your Mac
+
+* Ensure you have Xcode installed and have accepted the terms of use and launched Xcode at least once.
+* Install [Visual Studio for Mac ](https://visualstudio.microsoft.com/vs/mac). (This will also install the `dotnet` CLI).
+* If you still do not have the `dotnet` cli `>= v2.1.300` installed [manually install it](https://www.microsoft.com/net/download/dotnet-core/2.1)
+* [Install Homebrew](https://brew.sh/)
```
Comment (nickgra) [inline Readme.md]: This is all handled by the Prep script so this can be removed.

File: Readme.md [hunk 2]
```diff
@@ -46,15 +46,95 @@ If you'd like to build your own VFS for Git Windows installer:
...
+  ```
+  You may have the "XCode Command Line Tools" installed (helpfully by Mac OS) instead of full XCode.
+  Make sure
+  ```
+  xcode-select -p
+  ```
+
+  shows `/Applications/Xcode.app/Contents/Developer`. If it does not, install Xcode and then launch it (you can close it afterwards.)
+
+* Now setup the git credential manager by running
```
Comment (nickgra) [inline Readme.md]: I would put all this in the above section as it's part of the Build/Run steps, not troubleshooting :)

File: Readme.md [hunk 3]
```diff
@@ -46,15 +46,95 @@ If you'd like to build your own VFS for Git Windows installer:
...
+
+  You should see a path to the gvfs executable.
+
+* Try cloning! Now that you have `gvfs` ready you can try cloning a VFS for Git enabled repository!
+
+  ```
+  gvfs clone URL_TO_REPOSITORY [--cache-url] --local-cache-path ~/.gvfsCache
+  ```
+
+  Note you may have a cache server URL to use. Note the current use of `--local-cache-path`. That argument prevent gvfs from running into a permissions error trying to put the cache at the root of your Mac's hard-drive.
```
Comment (nickgra) [inline Readme.md]: Seems out of scope for people who are interested in hacking on the VFS For Git codebase rather than customers who want to run a VFS for Git repo on their machine.

File: Readme.md [hunk 4]
```diff
@@ -46,15 +46,95 @@ If you'd like to build your own VFS for Git Windows installer:
...
+* Install [Visual Studio for Mac ](https://visualstudio.microsoft.com/vs/mac). (This will also install the `dotnet` CLI).
+* If you still do not have the `dotnet` cli `>= v2.1.300` installed [manually install it](https://www.microsoft.com/net/download/dotnet-core/2.1)
+* [Install Homebrew](https://brew.sh/)
+* Install Java with
+  ```
+  brew cask install java
+  ```
+
+* Install the Git credential manager for Mac [by following their Homebrew instructions](https://github.com/Microsoft/Git-Credential-Manager-for-Mac-and-Linux/blob/master/Install.md#installing-on-mac-using-homebrew-or-on-linux-using-linuxbrew-recommended)
+
```
Comment (kyle-rader) [inline Readme.md]: @nickgra is referring to his comment down to here.

File: Readme.md (hunk 2 above)
Comment (kyle-rader) [reply Readme.md]: This is resolved by modifying the Troubleshooting section to be more inline.

File: Readme.md [hunk 5]
```diff
@@ -46,15 +46,90 @@ If you'd like to build your own VFS for Git Windows installer:
...
 build will fail, and the second and subsequent builds will succeed. This is because the build requires a prebuild code generation step.
 For details, see the build script in the previous step.
 
 The installer can now be found at `C:\Repos\VFSForGit\BuildOutput\GVFS.Installer\bin\x64\[Debug|Release]\SetupGVFS.<version>.exe`
 
+## Building VFS for Git on Mac
+
+VFS for macOS is still in progress. You can build it, but this will not create a macOS VFS installer the same way the current Windows build will.
+
+* Ensure you have Xcode installed and have accepted the terms of use and launched Xcode at least once.
```
Comment (nickgra) [inline Readme.md]: nit: Only use `and` one in this step. replace the first and with ,

File: ProjFS.Mac/Scripts/Build.sh [hunk 6]
```diff
@@ -8,11 +8,11 @@ fi
 SCRIPTDIR=$(dirname ${BASH_SOURCE[0]})
 SRCDIR=$SCRIPTDIR/../..
 ROOTDIR=$SRCDIR/..
-PACKAGES=$ROOTDIR/packages 
+PACKAGES=$ROOTDIR/packages
 
 PROJFS=$SRCDIR/ProjFS.Mac
 
-xcodebuild -sdk macosx10.13 -configuration $CONFIGURATION -workspace $PROJFS/PrjFS.xcworkspace build -scheme PrjFS -derivedDataPath $ROOTDIR/BuildOutput/ProjFS.Mac/Native || exit 1
```
Comment (sanoursa) [inline ProjFS.Mac/Scripts/Build.sh]: Why did you remove the sdk flag here? On my machine, this was required for it to build properly (back when I wrote the script anyway, not sure about today)

File: Readme.md [hunk 7]
```diff
@@ -46,15 +46,90 @@ If you'd like to build your own VFS for Git Windows installer:
...
-* You can also build in Visual Studio by opening `src\GVFS.sln` (do not upgrade any projects) and building. However, the very first 
+* You can also build in Visual Studio by opening `src\GVFS.sln` (do not upgrade any projects) and building. However, the very first
 build will fail, and the second and subsequent builds will succeed. This is because the build requires a prebuild code generation step.
 For details, see the build script in the previous step.
 
 The installer can now be found at `C:\Repos\VFSForGit\BuildOutput\GVFS.Installer\bin\x64\[Debug|Release]\SetupGVFS.<version>.exe`
 
+## Building VFS for Git on Mac
+
+VFS for macOS is still in progress. You can build it, but this will not create a macOS VFS installer the same way the current Windows build will.
```
Comment (sanoursa) [inline Readme.md]: We've been saying "VFS for Git on Mac", so stick with that for consistency. I'm open to feedback if it would be better to say "VFS for Git on macOS", but "VFS for macOS" is a whole different thing (https://developer.apple.com/library/archive/documentation/Darwin/Conceptual/KernelProgramming/Filesystem/Filesystem.html)

File: Readme.md [hunk 8]
```diff
@@ -46,15 +46,90 @@ If you'd like to build your own VFS for Git Windows installer:
...
+  ```
+  csrutil disable
+  ```
+  to disable SIP.
+
+* Install [Visual Studio for Mac ](https://visualstudio.microsoft.com/vs/mac). (This will also install the `dotnet` CLI).
+
+* If you still do not have the `dotnet` cli `>= v2.1.300` installed [manually install it](https://www.microsoft.com/net/download/dotnet-core/2.1)
+
+* If you are not currently an Apple Developer you will need to become one so that you can either use Microsoft certs or create your own certs for signing purposes. (If you are a Microsoft employee you must use your alias@microsoft.com when creating your Apple account.)
```
Comment (sanoursa) [inline Readme.md]: I would remove all of this and instead point to a link for managing certs in Xcode. There's no reason for anyone to use Microsoft certs, nor do we want to open that up. For local development, people just need to generate their own cert and sign their kext, and that is all.

File: Readme.md [hunk 9]
```diff
@@ -46,15 +46,90 @@ If you'd like to build your own VFS for Git Windows installer:
...
+
+* Create a `VFS` directory and Clone the VFS repo into a directory called `src` inside it:
```
Comment (sanoursa) [inline Readme.md]: There is no VFS repo :-). It's called VFSForGit.

File: Readme.md [hunk 10]
```diff
@@ -46,15 +46,90 @@ If you'd like to build your own VFS for Git Windows installer:
...
+  ```
+  mkdir VFS && cd VFS && git clone https://github.com/Microsoft/VFSForGit.git src && cd src
+  ```
+
+* Now `cd` back into your VFS src directory and run
+
+  ```
+  Scripts/Mac/BuildGVFSForMac.sh [Release]
```
Comment (sanoursa) [inline Readme.md]: Technically, the param is `[Debug|Release]`

File: Readme.md [hunk 11]
```diff
@@ -46,15 +46,90 @@ If you'd like to build your own VFS for Git Windows installer:
...
+  ```
+  You may have the "XCode Command Line Tools" installed (helpfully by Mac OS) instead of full XCode.
+  Make sure
+  ```
+  xcode-select -p
+  ```
+
+  shows `/Applications/Xcode.app/Contents/Developer`. If it does not, install Xcode and then launch it (you can close it afterwards.)
+
+* Now setup the git credential manager by running
```
Comment (sanoursa) [inline Readme.md]: This script installs the GCM and also installs the downloaded version of git

Comment (sanoursa) [reply Readme.md]: And updates your java version. Let's not mess up someone's machine setup without warning them.

File: Readme.md [hunk 12]
```diff
@@ -46,15 +46,90 @@ If you'd like to build your own VFS for Git Windows installer:
...
+  Scripts/Mac/PrepFunctionalTests.sh
+  ```
+
+* Now you have to load the ProjFS Kext.
+
+  ```
+  ProjFS.Mac/Scripts/LoadPrjFSKext.sh [Release]
+  ```
+
+* Now you can put your built gvfs program on your path. A simple way to do that is by adding `Path/to/VFS/Publish` to your path.
```
Comment (sanoursa) [inline Readme.md]: Style nit: You don't need to start every step with the word "Now" :-)

File: Readme.md [hunk 13]
```diff
@@ -46,15 +46,90 @@ If you'd like to build your own VFS for Git Windows installer:
...
+
+  You should see a path to the gvfs executable.
+
+* Try cloning! Now that you have `gvfs` ready you can try cloning a VFS for Git enabled repository!
+
+  ```
+  gvfs clone URL_TO_REPOSITORY --local-cache-path ~/.gvfsCache
+  ```
+
+  Note the current use of `--local-cache-path`. That argument prevent gvfs from running into a permissions error trying to put the cache at the root of your Mac's hard-drive.
```
Comment (sanoursa) [inline Readme.md]: ... because porting of this feature is still in progress

File: Scripts/Mac/PrepFunctionalTests.sh [hunk 14]
```diff
@@ -36,13 +36,6 @@ fi
 
 git-credential-manager install
 
-# If our Java version is 9+ (the formatting of 'java -version' changed in Java 9), work around
```
Comment (sanoursa) [inline Scripts/Mac/PrepFunctionalTests.sh]: Why did you remove all of this?

File: Readme.md [hunk 15]
```diff
@@ -46,15 +46,90 @@ If you'd like to build your own VFS for Git Windows installer:
...
 
 The installer can now be found at `C:\Repos\VFSForGit\BuildOutput\GVFS.Installer\bin\x64\[Debug|Release]\SetupGVFS.<version>.exe`
 
+## Building VFS for Git on Mac
+
+VFS for macOS is still in progress. You can build it, but this will not create a macOS VFS installer the same way the current Windows build will.
+
+* Ensure you have Xcode installed and have accepted the terms of use and launched Xcode at least once.
+
+* Disable "System Integrity Protection" (for loading unsigned kexts) by booting into recovery mode (`[Win/⌘] + R` while booting).
```
Comment (sanoursa) [inline Readme.md]: You can't load an unsigned kext, period. Our kexts are not unsigned, they are just signed with developer certs.

File: Readme.md [hunk 16]
```diff
@@ -46,15 +46,90 @@ If you'd like to build your own VFS for Git Windows installer:
...
+  Once booted into recovery mode open Utilities -> Terminal to launch a termnial. Enter:
```
Comment (sanoursa) [inline Readme.md]: Typo "termnial"

File: Readme.md [hunk 17]
```diff
@@ -46,15 +46,90 @@ If you'd like to build your own VFS for Git Windows installer:
...
+
+* Install [Visual Studio for Mac ](https://visualstudio.microsoft.com/vs/mac). (This will also install the `dotnet` CLI).
+
+* If you still do not have the `dotnet` cli `>= v2.1.300` installed [manually install it](https://www.microsoft.com/net/download/dotnet-core/2.1)
+
+* If you are not currently an Apple Developer you will need to become one so that you can either use Microsoft certs or create your own certs for signing purposes. (If you are a Microsoft employee you must use your alias@microsoft.com when creating your Apple account.)
+
+* Create a `VFS` directory and Clone the VFS repo into a directory called `src` inside it:
+  ```
+  mkdir VFS && cd VFS && git clone https://github.com/Microsoft/VFSForGit.git src && cd src
```
Comment (sanoursa) [inline Readme.md]: Please follow a similar style as the Windows section for the steps on how to clone and build.

Actually I'm not picky on what the style is, just that they be consistent.

File: ProjFS.Mac/Scripts/Build.sh (hunk 6 above)
Comment (nickgra) [reply ProjFS.Mac/Scripts/Build.sh]: Xcode 10 is out now, which removes the 10.13 SDK and replaces it with 10.14. So we can update all of our machines to 10 and update these strings to 10.14, or remove them altogether. I think we only needed the SDK flag in the old days when we were using functions that weren't in the 10.12 SDK. 

We also may have misconfigured our Xcode project in the past to not have a default SDK setup, but that is no longer the case :)

File: Scripts/Mac/PrepFunctionalTests.sh (hunk 14 above)
Comment (nickgra) [reply Scripts/Mac/PrepFunctionalTests.sh]: No longer needed now that 2.0.4 of the GCM is rolled out with the real fix :)

File: Readme.md (hunk 8 above)
Comment (kyle-rader) [reply Readme.md]: How is [this](https://developer.apple.com/documentation/security/certificate_key_and_trust_services/certificates) as the resource to link to?

File: Readme.md (hunk 10 above)
Comment (kyle-rader) [reply Readme.md]: Hmm, so with this script no arg is the same as passing Debug, I don't know the sort of POSIX way of specifying that (That you have two options here but one is chosen as the default?

File: Readme.md (hunk 8 above)
Comment (nickgra) [reply Readme.md]: That page seems too high level on what a certificate is (we should point them to codesigning resources using certificates), this might be a better starting point https://developer.apple.com/support/code-signing/ 

In general, Xcode should be able to do all the right things for them once they're signed in.

**Comment by kyle-rader** [comment]
@sanoursa @wilbaker  Updated and rebased on master. Ready for re-review

File: Readme.md [hunk 18]
```diff
@@ -46,15 +46,95 @@ If you'd like to build your own VFS for Git Windows installer:
...
 
 The installer can now be found at `C:\Repos\VFSForGit\BuildOutput\GVFS.Installer\bin\x64\[Debug|Release]\SetupGVFS.<version>.exe`
 
+## Building VFS for Git on Mac
+
+Note that VFS for Git on Mac is under active development.
+
+* Ensure you have Xcode installed, have accepted the terms of use, and have launched Xcode at least once.
+
+* Disable the "System Integrity Protection" (for loading unsigned Kexts) by booting into recovery mode (`[Win/⌘] + R` while booting).
```
Comment (sanoursa) [inline Readme.md]: There's no such thing as an unsigned kext. You can't even build it if you don't have your cert set up. What we do have are dev-signed kexts and production-signed kexts.

Also please don't capitalize kext.

Comment (sanoursa) [inline Readme.md]: We probably want a giant warning message here that this is a Very Bad Thing for the security of your machine. We are not recommending this on any production machine, ever. This is for development and testing only. Etc etc.

Comment (sanoursa) [inline Readme.md]: And for security purposes, we should put this as the final step _after_ all the steps of installing this and that. We don't want someone to disable SIP and then inadvertently install a malicious kext because we opened up their machine.

File: Readme.md [hunk 19]
```diff
@@ -46,15 +46,95 @@ If you'd like to build your own VFS for Git Windows installer:
...
+
+* Install [Visual Studio for Mac ](https://visualstudio.microsoft.com/vs/mac). (This will also install the `dotnet` CLI).
+
+* If you still do not have the `dotnet` cli `>= v2.1.300` installed [manually install it]. You can check what version you have with `dotnet --version`.(https://www.microsoft.com/net/download/dotnet-core/2.1)
+
+* You will need to manage and sign your own certificate. [Apple Developer Cert Docs](https://developer.apple.com/documentation/security/certificate_key_and_trust_services/certificates).
+
+* Create a `VFSForGit` directory and Clone VFSForGit into a directory called `src` inside it:
+  ```
+  mkdir VFSForGit && cd VFSForGit && git clone https://github.com/Microsoft/VFSForGit.git src && cd src
```
Comment (sanoursa) [inline Readme.md]: Please match the style of the Windows section up above. Spell out each step on its own line.

File: Readme.md (hunk 10 above)
Comment (sanoursa) [reply Readme.md]: I think `[Debug|Release]` is the correct way to specify this. You can specify either string, but the whole thing is optional. The fact that there is a default choice has no bearing on the syntax.

File: Readme.md [hunk 20]
```diff
@@ -46,15 +46,95 @@ If you'd like to build your own VFS for Git Windows installer:
...
+  * install/update Java (with `brew`)
+  * install a VFS for Git aware version of Git
+
+  ```
+  Scripts/Mac/PrepFunctionalTests.sh
+  ```
+
+* Now you have to load the ProjFS Kext.
+
+  ```
```
Comment (sanoursa) [inline Readme.md]: This is the first time that we actually require that SIP be disabled.

File: Readme.md [hunk 21]
```diff
@@ -46,15 +46,95 @@ If you'd like to build your own VFS for Git Windows installer:
...
+  You may have the "XCode Command Line Tools" installed (helpfully by Mac OS) instead of full XCode.
+  Make sure
+  ```
+  xcode-select -p
+  ```
+
+  shows `/Applications/Xcode.app/Contents/Developer`. If it does not, install Xcode and then launch it (you can close it afterwards.)
+
+* Prep your machine to use VFS for Git. The following are all done by the script below.wwwww
```
Comment (wilbaker) [inline Readme.md]: > below.wwwww

Remove "wwwww"

File: Readme.md [hunk 22]
```diff
@@ -46,15 +46,95 @@ If you'd like to build your own VFS for Git Windows installer:
...
+  * install Homebrew
+  * install and setup the git credential manager (with `brew`)
```
Comment (wilbaker) [inline Readme.md]: Nit: Capitalization "Git Credential Manager"

File: Readme.md [hunk 23]
```diff
@@ -46,15 +46,95 @@ If you'd like to build your own VFS for Git Windows installer:
...
+  * install a VFS for Git aware version of Git
+
+  ```
+  Scripts/Mac/PrepFunctionalTests.sh
+  ```
+
+* Now you have to load the ProjFS Kext.
+
+  ```
+  ProjFS.Mac/Scripts/LoadPrjFSKext.sh [Release]
```
Comment (wilbaker) [inline Readme.md]: Same comment that @sanoursa made above `[Debug|Release]`

File: Readme.md [hunk 24]
```diff
@@ -46,15 +46,95 @@ If you'd like to build your own VFS for Git Windows installer:
...
+  ```
+
+* Add your built gvfs program to your path. A simple way to do that is by adding `Path/to/VFS/Publish` to your path.
```
Comment (wilbaker) [inline Readme.md]: Minor suggestion:

> Add your built gvfs program to your path.

Everything so far has said "VFS for Git" and so seeing "gvfs" might be a big surprising.

You might want to reword this to something like:

> Add your built VFS for Git executable ("gvfs") to your path.

Comment (wilbaker) [inline Readme.md]: > adding `Path/to/VFS/Publish` to your path

In the steps above the name of the folder is `VFSForGit`, should this step be changed to something like:

> adding `Path/to/VFSForGit/Publish` to your path

or 

> adding `<Path to VFSForGit>/Publish` to your path

File: Readme.md [hunk 25]
```diff
@@ -46,15 +46,95 @@ If you'd like to build your own VFS for Git Windows installer:
...
+
+  You should see a path to the gvfs executable.
+
+* Try cloning a VFS for Git enabled repository!
+
+  ```
+  gvfs clone URL_TO_REPOSITORY --local-cache-path ~/.gvfsCache
+  ```
+
+  Note the current use of `--local-cache-path`. That argument prevents VFS for Git from running into a permissions error trying to put the cache at the root of your Mac's hard-drive because porting of this feature is still in progress.
```
Comment (wilbaker) [inline Readme.md]: Style comment.

I found this:

> Note the current use of `--local-cache-path`. That argument prevents VFS for Git from running into a permissions error trying to put the cache at the root of your Mac's hard-drive because porting of this feature is still in progress.

A bit tough to follow.  Consider splitting up the second sentence and adding a few more details:

> Note the current use of `--local-cache-path`. Without this argument VFS for Git will encounter a permissions error when it attempts to create its cache at the root of your hard-drive. Automatic picking of the cache path has not yet been ported to VFS for Git on Mac.

File: Readme.md [hunk 26]
```diff
@@ -46,15 +46,95 @@ If you'd like to build your own VFS for Git Windows installer:
...
+  ```
+  csrutil disable
+  ```
+  to disable SIP. Then click the Apple logo in the top left and restart.
+
+* Install [Visual Studio for Mac ](https://visualstudio.microsoft.com/vs/mac). (This will also install the `dotnet` CLI).
+
+* If you still do not have the `dotnet` cli `>= v2.1.300` installed [manually install it]. You can check what version you have with `dotnet --version`.(https://www.microsoft.com/net/download/dotnet-core/2.1)
+
+* You will need to manage and sign your own certificate. [Apple Developer Cert Docs](https://developer.apple.com/documentation/security/certificate_key_and_trust_services/certificates).
```
Comment (nickgra) [inline Readme.md]: This link isn't helpful to end users functionally need to just sign into Xcode, they don't need to learn about the APIs :)

On a fresh machine, I launched Xcode and opened the workspace. The build failed due to signing, clicking it took me to a page to sign in with my Apple ID which automatically fixed everything for me.

So I'd rephrase it like this:
If you're using Xcode for the first time, you may have to login to Xcode with your Apple ID to generate a codesigning certificate. 

or extra verbose:
If you're using Xcode for the first time, you may have to login to Xcode with your Apple ID to generate a codesigning certificate. You can do this by launching Xcode.app, opening the PrjFS.xcworkspace and trying to build. You can find the signing options in the General tab of the project's settings.

Comment (kyle-rader) [reply Readme.md]: I like your verbose option

**Comment by kyle-rader** [comment]
@nickgra @sanoursa  I've pushed resolutions to all remaining comments

File: Readme.md [hunk 27]
```diff
@@ -46,15 +46,111 @@ If you'd like to build your own VFS for Git Windows installer:
...
+  ```
+  You may have the "XCode Command Line Tools" installed (helpfully by Mac OS) instead of full XCode.
+  Make sure
+  ```
+  xcode-select -p
+  ```
+
+  shows `/Applications/Xcode.app/Contents/Developer`. If it does not, install Xcode and then launch it (you can close it afterwards.)
+
+* For the time being, only for active development, you will have to disable the SIP (System Integrity Protection) in order to load the kext).
```
[the remark this answers is not in the document]
Comment (nickgra) [reply Readme.md]: I'd just remove the 'For the time being, only for active development' here.

File: Readme.md [hunk 28]
```diff
@@ -46,15 +46,111 @@ If you'd like to build your own VFS for Git Windows installer:
...
+
+  **This is dangerous and very bad for the security of your machine. Do not do this on any production machine! If you no longer need to developer VFS for Git on Mac we recommend re-enabling SIP ASAP.**
```
[the remark this answers is not in the document]
Comment (kyle-rader) [reply Readme.md]: That's not a `nit` that's straight up incorrect grammar :)

File: Readme.md (hunk 27 above)
[the remark this answers is not in the document]
Comment (sanoursa) [reply Readme.md]: Not sure I agree. Most people will never need to rebuild the kext, just perhaps the user mode code. Lets err on the side of making people overly paranoid about disabling SIP.

[the remark this answers is not in the document]
Comment (nickgra) [reply Readme.md]: I agree that most people won't need/want to build the kext, but until we provide them with a properly codesigned PrjFSKext, they'll have to do that. IMO, whenever we get to the point where we publish a properly signed installer, we should update the docs.

[the remark this answers is not in the document]
Comment (sanoursa) [reply Readme.md]: Yes but I'd rather have fewer people doing that rather than have anyone running with SIP disabled on accident. The statement here is accurate - right now, you cannot even use VFS for Git without disabling SIP, but that is just a moment in time limitation because we're in active development. This is not the desired state, and will stop being the case as soon as we publish a production-signed kext.
